### PR TITLE
[JENKINS-67002] protect Jenkins against bad RestartListeners

### PR DESCRIPTION
**Pull request by jtnord**
Observed in the ATH that the MailWatcher plugin would sometimes throw an
exception that would leave Jenkins in a zombie state (the app is still
set to "Jenkins is restarting" and nothing changes this state

See [JENKINS-67002](https://issues.jenkins-ci.org/browse/JENKINS-67002).

### Proposed changelog entries

* An exception thrown by a `RestartListener` no longer leaves Jenkins in a zombie-like state.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by MRamonLeon — COMMENTED** [review]
😍 these enforcements. They look unneeded but they protect the instance. Thanks

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (jtnord) [reply core/src/main/java/jenkins/model/Jenkins.java:4471]: It’s already in the stack trace - do we want to add the same information again?

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
